gradle-plugin-testing/src/test/java/com/palantir/gradle/plugintesting/GradleTestVersionsConfigTest.java

```diff
@@ -0,0 +1,289 @@
+
+            SortedSet<String> extraVersions = new TreeSet<>();
+            extraVersions.add("8.8");
+            extraVersions.add("8.10");
+            extraVersions.add("9.0");
+
+            GradleTestVersionsConfig config = GradleTestVersionsConfig.builder()
+                    .majorVersions(majorVersions)
+                    .extraVersions(extraVersions)
+                    .build();
+
+            GradleTestVersionsConfig updated = config.withoutMajorVersion(8);
+
+            assertThat(updated.majorVersions()).doesNotContainKey(8);
+            assertThat(updated.extraVersions()).containsExactly("9.0");
+        }
+
+        @Test
+        public void keeps_unrelated_extra_versions() {
+            SortedMap<Integer, String> majorVersions = new TreeMap<>();
+            majorVersions.put(7, "7.6.4");
+            majorVersions.put(8, "8.14.2");
+
+            SortedSet<String> extraVersions = new TreeSet<>();
+            extraVersions.add("7.5");
+            extraVersions.add("8.8");
+
+            GradleTestVersionsConfig config = GradleTestVersionsConfig.builder()
+                    .majorVersions(majorVersions)
+                    .extraVersions(extraVersions)
+                    .build();
+
+            GradleTestVersionsConfig updated = config.withoutMajorVersion(8);
+
+            assertThat(updated.majorVersions()).containsEntry(7, "7.6.4");
+            assertThat(updated.extraVersions()).containsExactly("7.5");
+        }
+
+        @Test
+        public void handles_non_existent_major_versions() {
+            SortedMap<Integer, String> majorVersions = new TreeMap<>();
+            majorVersions.put(8, "8.14.2");
+
+            GradleTestVersionsConfig config = GradleTestVersionsConfig.builder()
+                    .majorVersions(majorVersions)
+                    .extraVersions(new TreeSet<>())
+                    .build();
+
+            GradleTestVersionsConfig updated = config.withoutMajorVersion(9);
+
+            assertThat(updated.majorVersions()).containsEntry(8, "8.14.2");
+            assertThat(updated.majorVersions()).hasSize(1);
+        }
+    }
+
+    @Nested
+    class Miscellaneous {
+        @Test
+        public void testBuilderPattern() {
+            SortedMap<Integer, String> majorVersions = new TreeMap<>();
+            majorVersions.put(8, "8.14.2");
+            majorVersions.put(9, "9.2.0");
+
+            SortedSet<String> extraVersions = new TreeSet<>();
+            extraVersions.add("8.8");
+
+            GradleTestVersionsConfig config = GradleTestVersionsConfig.builder()
+                    .majorVersions(majorVersions)
+                    .extraVersions(extraVersions)
+                    .build();
+
+            assertThat(config.majorVersions()).containsEntry(8, "8.14.2").containsEntry(9, "9.2.0");
+            assertThat(config.extraVersions()).containsExactly("8.8");
+        }
+
+        @Test
+        public void testEqualsAndHashCode() {
```

Comment:
you don't need to do this, immutables generates this for you and your previous tests will effectively check this for you because you're calling equals all the time